config/interfaces.go

```diff
@@ -0,0 +1,8 @@
+package config
+
+// PostProcess is the interface used to do any sort of processing after `config.Load` has been
+// called. This runs after the entire struct has been populated from the configuration files and
+// environment variables
+type PostProcess interface {
```

Comment:
suggestion: since the verb that this is invoked around is after "load" a more specific name could be `PostLoad`